pkg/gardenlet/bootstrap/bootstrap.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+This file was copied and modified from the kubernetes/kubernetes project
+https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/certificate/bootstrap/bootstrap.go
+
+Modifications Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved.
+*/
+
+package bootstrap
+
+import (
+	"crypto"
+	"crypto/sha512"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/base64"
+	"fmt"
+	"time"
+
+	certificatesv1beta1 "k8s.io/api/certificates/v1beta1"
+	certificatesv1beta1client "k8s.io/client-go/kubernetes/typed/certificates/v1beta1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	certutil "k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/certificate/csr"
+	"k8s.io/client-go/util/keyutil"
+)
+
+// RequestSeedCertificate will create a certificate signing request for a seed
+// (Organization and CommonName for the CSR will be set as expected for seed
+// certificates) and send it to API server, then it will watch the object's
+// status, once approved by API server, it will return the API server's issued
+// certificate (pem-encoded). If there is any errors, or the watch timeouts, it
+// will return an error. This is intended for use on seeds (gardenlet).
+func RequestSeedCertificate(client certificatesv1beta1client.CertificateSigningRequestInterface, privateKeyData []byte, seedNames string) (certData []byte, err error) {
+	subject := &pkix.Name{
+		Organization: []string{"gardener.cloud:system:seeds"},
+		CommonName:   "gardener.cloud:system:seed:" + string(seedNames), // TODO
```

Comment:
Is it really intended for multiple `seedNames` and what's the TODO about?

Reply:
sorry, left-over, thanks for finding it. if a gardenlet targets more than one seed then the `seedName` will be `<ambigious>`.